pkg/controller/s2irun/s2irun_controller.go

```diff
@@ -224,7 +233,126 @@ func (r *ReconcileS2iRun) Reconcile(request reconcile.Request) (reconcile.Result
+							if reflectutils.Contains(container.Name, scale.Containers) {
+								if deploy.Spec.Template.Spec.Containers[0].Image == builder.Spec.Config.Tag {
+									deploy.Spec.Template.Spec.Containers[0].ImagePullPolicy = corev1.PullAlways
+								} else {
+									deploy.Spec.Template.Spec.Containers[0].Image = builder.Spec.Config.Tag
+								}
+							}
+						}
+					}
+					if deploy.Spec.Template.Labels == nil {
+						deploy.Spec.Template.Labels = make(map[string]string)
+					}
+
+					deploy.Spec.Template.Labels[devopsv1alpha1.WorkloadLatestS2iRunTemplateLabel] = instance.Name
+
+					log.Info("Update deployment", "ns", deploy.Namespace, "statefulSet", deploy.Name)
+					if err := r.Update(context.TODO(), deploy); err != nil {
+						errs = append(errs, err)
+						continue
+					}
+					completedScaleWorkloads = append(completedScaleWorkloads, scale)
+
+				case devopsv1alpha1.KindStatefulSet:
+					statefulSet := &v1.StatefulSet{}
+					err := r.Get(context.TODO(), types.NamespacedName{Namespace: instance.Namespace, Name: scale.Name}, statefulSet)
+					if err != nil {
+						errs = append(errs, err)
+						continue
+					}
+					log.Info("Autoscale StatefulSet", "ns", instance.Namespace, "statefulSet", statefulSet.Name)
+					if len(statefulSet.Spec.Template.Spec.Containers) == 1 {
+						if statefulSet.Spec.Template.Spec.Containers[0].Image == builder.Spec.Config.Tag {
+							statefulSet.Spec.Template.Spec.Containers[0].ImagePullPolicy = corev1.PullAlways
+						} else {
+							statefulSet.Spec.Template.Spec.Containers[0].Image = builder.Spec.Config.Tag
+						}
+					} else {
+						for _, container := range statefulSet.Spec.Template.Spec.Containers {
+							if reflectutils.Contains(container.Name, scale.Containers) {
+								if statefulSet.Spec.Template.Spec.Containers[0].Image == builder.Spec.Config.Tag {
+									statefulSet.Spec.Template.Spec.Containers[0].ImagePullPolicy = corev1.PullAlways
+								} else {
+									statefulSet.Spec.Template.Spec.Containers[0].Image = builder.Spec.Config.Tag
+								}
+							}
+						}
+					}
+					if statefulSet.Spec.Template.Labels == nil {
+						statefulSet.Spec.Template.Labels = make(map[string]string)
+					}
+
+					statefulSet.Spec.Template.Labels[devopsv1alpha1.WorkloadLatestS2iRunTemplateLabel] = instance.Name
+
+					log.Info("Update statefusSet", "ns", statefulSet.Namespace, "statefulSet", statefulSet.Name)
+					if err := r.Update(context.TODO(), statefulSet); err != nil {
+						errs = append(errs, err)
+						continue
+					}
+					completedScaleWorkloads = append(completedScaleWorkloads, scale)
+				default:
+					errs = append(errs, fmt.Errorf("unsupport workload Kind [%s], name [%s]", scale.Kind, scale.Name))
+				}
+			}
+			if completedScaleAnnotation, err := json.Marshal(completedScaleWorkloads); err != nil {
+				return reconcile.Result{}, err
+			} else {
+				origin.Annotations[devopsv1alpha1.S2irCompletedScaleAnnotations] = string(completedScaleAnnotation)
```

Comment:
`origin` is used to save current state of `s2irun`. We should not change `origin`. change `instance` and compare with origin to `deside` whether we need an update